sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -186,6 +190,13 @@ class CodegenContext {
       freshNameIds += fullName -> 1
       fullName
     }
+    freshNameJavaTypes.put(freshname, javaType)
+    freshname
+  }
+
+  def isColumnarType(name: String): Boolean = {
```

Comment:
Is this a more explicit way to maintain this? For example, in PhysicalRDD when scanBatches() is generated, we set is columnarInput to true. Looking this up by name is brittle.